packages/desktop-client/src/components/modals/LoadBackup.jsx

```diff
@@ -55,10 +56,12 @@ export function LoadBackup({
   modalProps,
 }) {
   const [backups, setBackups] = useState([]);
+  const prefsBudgetId = useLocalPref('id');
+  const budgetIdToLoad = budgetId || prefsBudgetId;
 
   useEffect(() => {
-    send('backups-get', { id: budgetId }).then(setBackups);
-  }, [budgetId]);
+    send('backups-get', { id: budgetIdToLoad }).then(setBackups);
+  }, [budgetIdToLoad]);
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏why do we need this functional change?

Reply:
Originally, the budget id was being retrieved by the root `Modals` component and passed to this component. I just moved that retrieval logic to this component.